crates/web/src/lib.rs

```diff
@@ -59,10 +59,216 @@ impl<'a> rose::FuncNode for &'a Func {
+                    write!(&mut s, "[")?;
+                    print_elems(s, elems)?;
+                    writeln!(&mut s, "]")?;
+                }
+                rose::Expr::Tuple { members } => {
+                    write!(&mut s, "(")?;
+                    print_elems(s, members)?;
+                    writeln!(&mut s, ")")?;
+                }
+                rose::Expr::Index { array, index } => {
+                    writeln!(&mut s, "x{}[x{}]", array.var(), index.var())?
+                }
+                rose::Expr::Member { tuple, member } => {
+                    writeln!(&mut s, "x{}.{}", tuple.var(), member.member())?
+                }
+                rose::Expr::Slice { array, index } => {
+                    writeln!(&mut s, "x{}![x{}]", array.var(), index.var())?
+                }
+                rose::Expr::Field { tuple, field } => {
+                    writeln!(&mut s, "x{}!.{}", tuple.var(), field.member())?
+                }
+                rose::Expr::Unary { op, arg } => match op {
+                    rose::Unop::Not => writeln!(&mut s, "not x{}", arg.var())?,
+                    rose::Unop::Neg => writeln!(&mut s, "-x{}", arg.var())?,
+                    rose::Unop::Abs => writeln!(&mut s, "|x{}|", arg.var())?,
+                    rose::Unop::Sqrt => writeln!(&mut s, "sqrt(x{})", arg.var())?,
+                },
+                rose::Expr::Binary { op, left, right } => match op {
+                    rose::Binop::And => writeln!(&mut s, "x{} and x{}", left.var(), right.var())?,
+                    rose::Binop::Or => writeln!(&mut s, "x{} or x{}", left.var(), right.var())?,
+                    rose::Binop::Iff => writeln!(&mut s, "x{} iff x{}", left.var(), right.var())?,
+                    rose::Binop::Xor => writeln!(&mut s, "x{} xor x{}", left.var(), right.var())?,
+                    rose::Binop::Neq => writeln!(&mut s, "x{} != x{}", left.var(), right.var())?,
+                    rose::Binop::Lt => writeln!(&mut s, "x{} < x{}", left.var(), right.var())?,
+                    rose::Binop::Leq => writeln!(&mut s, "x{} <= x{}", left.var(), right.var())?,
+                    rose::Binop::Eq => writeln!(&mut s, "x{} == x{}", left.var(), right.var())?,
+                    rose::Binop::Gt => writeln!(&mut s, "x{} > x{}", left.var(), right.var())?,
+                    rose::Binop::Geq => writeln!(&mut s, "x{} >= x{}", left.var(), right.var())?,
+                    rose::Binop::Add => writeln!(&mut s, "x{} + x{}", left.var(), right.var())?,
+                    rose::Binop::Sub => writeln!(&mut s, "x{} - x{}", left.var(), right.var())?,
+                    rose::Binop::Mul => writeln!(&mut s, "x{} * x{}", left.var(), right.var())?,
+                    rose::Binop::Div => writeln!(&mut s, "x{} / x{}", left.var(), right.var())?,
+                },
+                rose::Expr::Call { func, arg } => {
+                    writeln!(&mut s, "F{}(x{})", func.func(), arg.var())?
```

Comment:
Super minor; maybe we could distinguish `Func` from `Function` by pprinting them as lowercase and uppercase, respectively?

```suggestion
                    writeln!(&mut s, "f{}(x{})", func.func(), arg.var())?
```